OSDOCS-19851: adds disconnected install docs RHCL (#112599)
openshift/openshift-docs: modules/proc-install-on-openshift-cli-cio.adoc

Reviewer: ```suggestion
. Create a {prodname} custom resource (CR) that is similar to the following example CR:
```

  name: rhcl-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
----
+
* Replace `_<kuadrant_system>_` with the name of the namespace where you want to deploy {prodname}.
+
[IMPORTANT]
====
For disconnected installations, you must replace the `spec.source` parameter value of the `Subscription` object with the name of the `CatalogSource` object created by `oc-mirror`. You can find the catalog source name by running the following command:

[source,terminal]
----
$ oc get catalogsource -n openshift-marketplace
----
====

. Apply the `Subscription` CR by running the following command:
+
[source,terminal]
----
$ oc apply -f subscription.yaml
----

. Create the `OperatorGroup` custom resource (CR) by using the following example:
+
[source,terminal,subs="+quotes"]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: kuadrant
  namespace: _<kuadrant_system>_
spec:
  upgradeStrategy: Default
----
+
* Replace `_<kuadrant_system>_` with the namespace you used.

> * For disconnected installations, set the `spec` parameter to empty, `spec: {}`.

Reviewer 2: I would remove this line. How operator upgrades work doesn't change on disconnected installs, and `upgradeStrategy` probably still will be set to `Default` if `spec` is empty

Reviewer 3: Yeah tbh, install and upgrades should be no different after you create the subscription all going well. We probably want to avoid creating too many duplicate docs.

Author: this is the same module used in two places; no duplication of the main doc itself--just easier here not to click a link back to this in the regular install doc

  name: rhcl-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
----
+
* Replace `_<kuadrant_system>_` with the name of the namespace where you want to deploy {prodname}.
+
[IMPORTANT]
====
For disconnected installations, you must replace the `spec.source` parameter value of the `Subscription` object with the name of the `CatalogSource` object created by `oc-mirror`. You can find the catalog source name by running the following command:

[source,terminal]
----
$ oc get catalogsource -n openshift-marketplace
----
====

. Apply the `Subscription` CR by running the following command:
+
[source,terminal]
----
$ oc apply -f subscription.yaml
----

. Create the `OperatorGroup` custom resource (CR) by using the following example:
+
[source,terminal,subs="+quotes"]
----

> apiVersion: operators.coreos.com/v1alpha1

Reviewer 2: `apiVersion` key already exists, and `/v1` is the correct one

$ oc wait kuadrant/kuadrant --for="condition=Ready=true" -n _<kuadrant_system>_ --timeout=300s
----
+
Replace `_<kuadrant_system>_` with the namespace you used.
+
.Example output
[source,terminal,subs="+quotes"]
----
kuadrant.kuadrant.io/kuadrant Ready
----

> . Verify that all component Operator pods are running by entering the following command:

Reviewer: Formatting is off here:

<img width="1376" height="866" alt="Image" src="https://github.com/user-attachments/assets/6b3c5782-9bfd-486a-abc0-6f74a2254bfe" />

$ oc wait kuadrant/kuadrant --for="condition=Ready=true" -n _<kuadrant_system>_ --timeout=300s
----
+
Replace `_<kuadrant_system>_` with the namespace you used.
+
.Example output
[source,terminal,subs="+quotes"]
----
kuadrant.kuadrant.io/kuadrant Ready
----

. Verify that all component Operator pods are running by entering the following command:
+
[source,terminal]
----
$ oc get pods -n _<kuadrant_system>_
----
+
.Example output
[source,terminal]
----
NAME                                                READY   STATUS    RESTARTS   AGE
authorino-operator-controller-manager-<hash>        2/2     Running   0          5m
dns-operator-controller-manager-<hash>              2/2     Running   0          5m
kuadrant-operator-controller-manager-<hash>         2/2     Running   0          7m
limitador-operator-controller-manager-<hash>        2/2     Running   0          5m
----

. For a disconnected installation, verify that there are no image pull errors by running the following command:
+
[source,terminal,subs="+quotes"]
----

> $ oc get events -n kuadrant-system --field-selector reason=Failed

Reviewer 3: ```suggestion
$ oc get events -n _<kuadrant_system>_ --field-selector reason=Failed
```

$ oc wait kuadrant/kuadrant --for="condition=Ready=true" -n _<kuadrant_system>_ --timeout=300s
----
+
Replace `_<kuadrant_system>_` with the namespace you used.
+
.Example output
[source,terminal,subs="+quotes"]
----
kuadrant.kuadrant.io/kuadrant Ready
----

. Verify that all component Operator pods are running by entering the following command:
+
[source,terminal]
----

> $ oc get pods -n _<kuadrant_system>_

Reviewer 2: missing `+quotes` parameter

----
+
* Replace `_<kuadrant_system>_` with the namespace you used.
+
.Example output
[source,terminal]
----
NAME                                                READY   STATUS    RESTARTS   AGE
authorino-operator-controller-manager-<hash>        2/2     Running   0          5m
dns-operator-controller-manager-<hash>              2/2     Running   0          5m
kuadrant-operator-controller-manager-<hash>         2/2     Running   0          7m
limitador-operator-controller-manager-<hash>        2/2     Running   0          5m
----

. For a disconnected installation, verify that there are no image pull errors by running the following command:
+
[source,terminal,subs="+quotes"]
----
$ oc get events -n _<kuadrant_system>_ --field-selector reason=Failed
----
+
* Replace `_<kuadrant_system>_` with the namespace you used.
+
The output should show no image pull failures.
+
[NOTE]
====

> Pod specifications still show original image references, for example, `quay.io/kuadrant/...`, even when images are pulled from your mirror registry. The redirection happens transparently at the CRI-O level through `ImageDigestMirrorSet` or `ImageTagMirrorSet` configuration objects.

Reviewer 2: should we use the downstream example of the image instead?
```suggestion
Pod specifications still show original image references, for example, `registry.redhat.io/redhat/...`, even when images are pulled from your mirror registry. The redirection happens transparently at the CRI-O level through `ImageDigestMirrorSet` or `ImageTagMirrorSet` configuration objects.
```
